tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -335,6 +341,35 @@ pub fn logic_layer_aggregation(
 }
 
 
+// Genrates the source data/ annotaion of the cube for which the query is executed
+fn generate_source_data(cube: &Cube) -> Option<SourceMetadata> {
```

Comment:
The way this is written, this method will always return SourceMetadata, so it doesn't need to be a return type of`Option<SourceMetadata>` here. (You can wrap the response with `Some(x)`as needed?